api.go

```diff
@@ -95,47 +100,71 @@ func writeDatabaseError(w http.ResponseWriter) {
 	writeError(w, http.StatusInternalServerError, API_ERR_INTERNAL, "database error")
 }
 
-func distributeEvents() {
-	subscribers := make(map[chan string]struct{})
-	mut := sync.Mutex{}
-	msg := ""
-	debouncer := func() {
-		select {
-		case <-time.After(debounceDuration):
-			mut.Lock()
-			subs := subscribers
-			subscribers = make(map[chan string]struct{})
-			m := msg
-			msg = ""
-			incrementETag()
-			mut.Unlock()
-			log.Debugf("Delivering deployment change %s to %d subscribers", m, len(subs))
-			for subscriber := range subs {
-				select {
-				case subscriber <- m:
-					log.Debugf("Handling deploy response for: %s", m)
-					log.Debugf("delivering TO: %v", subscriber)
-				default:
-					log.Debugf("listener too far behind, message dropped")
+func debounce(in chan interface{}, out chan []interface{}, window time.Duration) {
+	send := func(toSend []interface{}) {
+		if toSend != nil {
+			log.Debugf("debouncer sending: %v", toSend)
+			out <- toSend
+		}
+	}
+	for {
+		incoming, ok := <-in
+		if !ok {
+			log.Debugf("closing debouncer")
+			close(out)
+			return
+		}
+		log.Debugf("debouncing %v", incoming)
+		toSend := []interface{}{incoming}
```

Comment:
no need for the initialization, go will handle `append(nil_slice, new_element)` automatically, please see my code in a latter comment.